server/consumer.go

```diff
@@ -5291,6 +5271,46 @@ func (o *consumer) stopWithFlags(dflag, sdflag, doSignal, advisory bool) error {
 	return err
 }
 
+// We need to optionally remove all messages since we are interest based retention.
+// We will do this consistently on all replicas. Note that if in clustered mode the non-leader
+// consumers will need to restore state first.
+// delete marks whether the consumer is being deleted, and should not be counted against interest.
+// No lock held on entry.
+func (o *consumer) cleanupNoInterestMessages(mset *stream, delete bool) {
+	state := mset.state()
+	stop := state.LastSeq
+	o.mu.Lock()
+	if !o.isLeader() {
+		o.readStoredState(stop)
+	}
+	start := o.asflr
+	o.mu.Unlock()
+	// Make sure we start at worst with first sequence in the stream.
+	if start < state.FirstSeq {
+		start = state.FirstSeq
+	}
+
+	// Ignore consumer if deleting, don't ignore if updating.
```

Comment:
This comment seems opposite to the code?

Reply:
When looking at just this snippet that's indeed true.

Since `mset.noInterest(seq, o)` ignores the supplied consumer by default, the logic is inverted.

Have updated the comment to be `Consumer's interests are ignored by default. If we should not ignore interest, unset.`
Hope that reflects the behavior better?